internal/maven/artifact.go

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright 2022.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package maven
+
+import (
+	"crypto/sha256"
+	hex2 "encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type Artifact struct {
+	root     string
+	fileName string
+}
+
+// this is useless since md5 broken?
+func (a *Artifact) GetMd5() (string, error) {
+	mu := a.root + "/" + a.fileName + ".md5"
+	if b, err := getRemote(mu); err != nil {
+		return "", err
+	} else {
+		return string(b), nil
+	}
+}
+
+// this is useless since sha1 broken?
+func (a *Artifact) GetSha1() (string, error) {
+	s1u := a.root + "/" + a.fileName + ".sha1"
+	if b, err := getRemote(s1u); err != nil {
+		return "", err
+	} else {
+		return string(b), nil
+	}
+}
+
+// returns raw bytes (not really consistent with above)
+func (a *Artifact) GetSha256() (string, error) {
+	b, err := getRemote(a.root + "/" + a.fileName)
+	if err != nil {
+		return "", err
+	}
+	h := sha256.Sum256(b)
+	hex := hex2.EncodeToString(h[:])
+
+	return hex, nil
+}
+
+func getRemote(url string) ([]byte, error) {
+	// should I be using that weird get with context so it's cancellable?
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("GET failed with status:%v", resp.StatusCode)
+	}
+	return ioutil.ReadAll(resp.Body)
```

Comment:
You know exactly how large this file should be -- either check that `ContentLength` matches or use a LimitedReader: https://golang.org/pkg/io/#LimitedReader